### PR TITLE
[TTD] Add Native Audience Support

### PR DESCRIPTION
**Pull request by maryamsharif**
This PR addresses S[TRATCONN-2990](https://segment.atlassian.net/jira/software/c/projects/STRATCONN/boards/310?modal=detail&selectedIssue=STRATCONN-2990&assignee=620ae2beef2bd5006ae88efb) to add native audience methods to The Trade Desk Crm.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)

Local Test Create Audience
```
curl --location 'http://localhost:3000/createAudience' \
--header 'Content-Type: application/json' \
--data '{
        "settings": {
                "authToken": "REDACTED",
                "advertiser_id": "REDACTED"
        },
        "audienceSettings": {
                "region": "US"
        },
        "audienceName": "Testing NEW Create Audience Method"     
}'

{"externalId":"tnlsw2x"}%
```

Local Test Get Audience
```
curl --location 'http://localhost:3000/getAudience' \
--header 'Content-Type: application/json' \
--data '{
        "settings": {
                "auth_token": "REDACTED",
                "advertiser_id": "REDACTED"
        },
        "externalId": "tnlsw2x"
}'

{"externalId":"tnlsw2x"}%       
```
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/the-trade-desk-crm/functions.ts [hunk 1]
```diff
@@ -64,6 +65,12 @@ export async function processPayload(input: ProcessPayloadInput) {
     // LEGACY FLOW
     // -----------
 
+    if (input.payloads.length < TTD_MIN_RECORD_COUNT) {
+      throw new PayloadValidationError(
+        `received payload count below The Trade Desk's ingestion limits. Expected: >=${TTD_MIN_RECORD_COUNT} actual: ${input.payloads.length}`
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/the-trade-desk-crm/functions.ts:70]: Nit:
```suggestion
        `received payload count below The Trade Desk's ingestion minimum. Expected: >=${TTD_MIN_RECORD_COUNT} actual: ${input.payloads.length}`
```

File: packages/destination-actions/src/destinations/the-trade-desk-crm/index.ts [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
+      SegmentName: string
+      Region: string
+      FirstPartyDataId: number
+    }
+  ]
+  PagingToken: string
+}
+
+const destination: AudienceDestinationDefinition<Settings, AudienceSettings> = {
+  name: 'The Trade Desk Crm',
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/the-trade-desk-crm/index.ts:40]: Should CRM be capitalized?

File: packages/destination-actions/src/destinations/the-trade-desk-crm/properties.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
 import { InputField } from '@segment/actions-core/destination-kit/types'
 
+export const external_id: InputField = {
+  label: 'External Audience ID',
+  description: 'The CRM Data ID for The Trade Desk Segment. ',
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/the-trade-desk-crm/properties.ts:5]:
```suggestion
  description: 'The CRM Data ID for The Trade Desk Segment.',
```

**Review by marinhero — CHANGES_REQUESTED** [review]
Left a few comments and questions. Overall structure of the audience destination looks good, we just gotta tighten some nuts and bolts. Good job!

File: packages/destination-actions/src/destinations/the-trade-desk-crm/syncAudience/__tests__/index.test.ts [hunk 4]
```diff
@@ -241,7 +244,7 @@ describe('TheTradeDeskCrm.syncAudience', () => {
     ).rejects.toThrow('Multiple audiences found with the same name')
   })
 
-  it('should fail if batch has less than 1500', async () => {
+  it('should fail if batch has less than 1500 and old flow', async () => {
```
[the remark this answers is not in the document]
Comment (maryamsharif) [reply packages/destination-actions/src/destinations/the-trade-desk-crm/syncAudience/__tests__/index.test.ts]: We moved the error message so it only errors out if we are using the legacy flow which sends events straight to TTD.

File: packages/destination-actions/src/destinations/the-trade-desk-crm/index.ts [hunk 5]
```diff
@@ -41,6 +74,86 @@ const destination: DestinationDefinition<Settings> = {
...
+      })
+
+      return {
+        externalId: response.data.CrmDataId
+      }
+    },
+    async getAudience(request, getAudienceInput) {
+      const crmDataId = getAudienceInput.externalId
+      const advertiserId = getAudienceInput.settings.advertiser_id
+      const authToken = getAudienceInput.settings.auth_token
```
[the remark this answers is not in the document]
Comment (maryamsharif) [reply packages/destination-actions/src/destinations/the-trade-desk-crm/index.ts]: No oauth, it is stored as a setting.

File: packages/destination-actions/src/destinations/the-trade-desk-crm/index.ts [hunk 6]
```diff
@@ -41,6 +74,86 @@ const destination: DestinationDefinition<Settings> = {
...
+
+      if (audienceName?.length == 0) {
+        throw new IntegrationError('Missing audience name value', 'MISSING_REQUIRED_FIELD', 400)
+      }
+
+      const response: ModifiedResponse<CREATE_API_RESPONSE> = await request(`${BASE_URL}/crmdata/segment`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'TTD-Auth': authToken
```
[the remark this answers is not in the document]
Comment (maryamsharif) [reply packages/destination-actions/src/destinations/the-trade-desk-crm/index.ts]: Moved to `extendRequest`.

**Review by marinhero — APPROVED** [review]

**Pull request merged**